src/components/TransactionItemNew/TransactionItem.js

```diff
@@ -0,0 +1,168 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+import classNames from 'classnames'
+import { connect } from 'react-redux'
+import {
+  memoize,
+  noop,
+} from 'lodash-es'
+
+import {
+  JAssetSymbol,
+  JIcon,
+  JLink,
+} from 'components/base'
+import { type Props as JIconProps } from 'components/base/JIcon/JIcon'
+import formatAssetBalance from 'utils/formatters/formatAssetBalance'
+
+import offsetsStyle from 'styles/offsets.m.scss'
+
+import {
+  type TransactionItem as TransactionItemRecord,
+  type TransactionDirection,
+  type TransactionStatus,
+  transactionsIndex,
+} from './transactionsIndex'
+
+import transactionItemStyle from './transactionItem.m.scss'
+
+const memoizedIndex = memoize(transactionsIndex)
+
+export type ContainerProps = {
+  +onClick?: (Address) => mixed,
+  +txAddress: Address,
+  +offset?: OffsetVariant,
+  +isActive?: boolean,
+}
+
+type Props = ContainerProps & {
+  +transaction: TransactionItemRecord,
+}
+
+const transactionIconMap: { [TransactionStatus | TransactionDirection]: JIconProps } = {
+  pending: {
+    name: 'clock-use-fill',
+    color: 'gray',
+  },
+  in: {
+    name: 'transaction-in-use-fill',
+    color: 'blue',
+  },
+  out: {
+    name: 'transaction-out-use-fill',
+    color: 'blue',
+  },
+  fail: {
+    name: 'circle-cross-use-fill',
+    color: 'red',
+  },
+  stuck: {
+    name: 'circle-alert-use-fill',
+    color: 'red',
+  },
+}
+
+function getTransactionIcon(
+  type: TransactionDirection,
+  status: TransactionStatus,
+): JIconProps {
+  if (status === 'success') {
+    return transactionIconMap[type]
+  }
+
+  return transactionIconMap[status]
+}
+
+function formatTransactionAmount({
+  asset,
+  amount,
+  type,
+  status,
+}: TransactionItemRecord): string {
+  const formatted = asset
+    ? formatAssetBalance(asset.blockchainParams.address, amount, asset.blockchainParams.decimals)
+    : amount
+  const symboled = `${formatted}\u202F${asset.symbol}`
+
+  return status === 'success' || status === 'pending'
+    ? `${(type === 'in' ? '+' : '\u2212')}${symboled}`
+    : symboled
+}
+
+class TransactionItem extends PureComponent<Props, *> {
+  static defaultProps = {
+    onClick: noop,
+  }
+
+  handleClick = (event: SyntheticEvent<HTMLAnchorElement>): void => {
+    event.preventDefault()
+
+    if (!this.props.isActive) {
+      // $FlowFixMe
```

Comment:
?

Reply:
Flow don't understand that `onClick?: (Address) => mixed` property with default value `noop` from `lodash` always is function.